File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.cpp
class Solution {
public:
    vector<int> ans;
    int dig;
    void solve(long x,int k,int n){
        if(dig==n){
            ans.push_back(x);
        }
        if(dig<n)
            return;
        
        
        int next = (x%10)+k,next2=(x%10)-k;
        if(next<10 and next>-1)
            solve(x*10 + next,k,n+1);
        if(next2<10 and next2>-1 and next!=next2)
            solve(x*10 + next2,k,n+1);
    }
    
    vector<int> numsSameConsecDiff(int n, int k) {
        dig=n;
        for(long i=1;i<10;i++) solve(i,k,1);
        return ans;
    }
};

File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.java
class Solution {
    ArrayList<Integer> arr = new ArrayList<Integer>();
    int lim,diff;
    void solve(int num,int digits){
        if(digits==lim){
            arr.add(num);
            return;
        }
        
        if(digits>lim) return;
        
        int nex = num%10+diff, nex2 = num%10-diff;
        
        if(nex>-1 && nex<10)
            solve(num*10 + nex,digits+1);
        
        if(nex2>-1 && nex2<10 && nex!=nex2)
            solve(num*10 + nex2,digits+1);
    }
    
    public int[] numsSameConsecDiff(int n, int k) {
        lim = n;
        diff = k; 
        for(int i=1;i<10;i++)
            solve(i,1);
        for(int a:arr) System.out.print(a);
        return arr.stream().mapToInt(i -> i).toArray();
    }
}